synapse/handlers/room.py

```diff
@@ -554,7 +555,9 @@ def create_room(self, requester, config, ratelimit=True, creator_join_profile=No
         invite_list = config.get("invite", [])
         for i in invite_list:
             try:
-                UserID.from_string(i)
+                uid = UserID.from_string(i)
+                # TODO Maybe attempt idna encoding/decoding as well?
```

Comment:
@richvdh Do you have any thoughts on this?

Reply:
I do have thoughts on it. https://matrix.org/docs/spec/appendices#server-name is clear that server_names (and hence mxids) may contain only ascii: it's up to the client to do idna-encoding/decoding on them if it so desires. 

Please can we get rid of the todo!

Reply:
```suggestion
```